module_utils/certificate/providers/certmonger.py

```diff
@@ -136,6 +136,18 @@ def request_certificate(self):
         # Set Domains
         for dns in self.csr.dns:
             command += ["-D", dns]
+        if not self.csr.dns:
+            command += ["-D", ""]
```

Comment:
this seems odd - why wouldn't you just omit the `-D` argument?  Or does that have a different semantic meaning than `-D ""`?

Reply:
It's indeed weird but it's required. 

For example, if we have 2 runs with the following settings:
First:
```
    certificate_requests:                                                                                 
      - name: mycert                                                                                      
        dns:                                                                                              
          - www.example.com                                                                                                                                    
        ca: self-sign    
```
Second:
```                                                                                
    certificate_requests:                                                                                 
      - name: mycert                                                                                      
        ip:                                                                                               
          - 192.0.2.12                                                                                         
        ca: self-sign    
```

If I don't set `-D ""` certmonger will leave `www.example.com` in the certificate which I don't think it's the behavior we want the role to have.